### ISSUE TITLE
 Feature: the ability to add to bookmark from thumbnail gallery without entering the image page.

### ISSUE THREAD

**Issue by Pandaa-P** [ppixiv/ppixiv#22]
It would be fantastic to be able to add an image to bookmark without clicking on the the image and entering its page to do that. 
I noticed that there is a light-bulb on every thumbnail so maybe add a heart next to it like the original site interface? or maybe the ability to do that by keyboard shortcut. 
`b` doesn't work from the thumbnail gallery and require the user to enter each image separately.  so maybe hover over the image and pressing `b` is possible to implement?

**Comment by ppixiv** [comment]
Right-click on a thumbnail in the search.  You can quickly bookmark and like images from the popup menu.

**Comment by Pandaa-P** [comment]
I tested your idea but the popup menu is too fast, I think it's a bug. It show up and in a less than a second disappeared like a flash, or maybe the issue is from using my laptop's trackpad and the touch is pain in the ass when dealing with Right-click and Left-click compilation which in the case of people with laptops is touch-based system.
I really appreciate if I can hover over an image and click a `b` to bookmark it. please if you have the time and willing to implement it please do. I have zero knowledge in programming aside from the absolute basic and really too dump to know any of JavaScript and do it by myself.
Thank you very much for your time and this amazing script

**Comment by ppixiv** [comment]
I did some rework on the context menu and hotkeys.  The context menu can be opened by holding ctrl, which might be easier for touchpads.  The hotkeys that were available in the image UI also work when the context menu is open now, so you can also ^B to bookmark while hovering over search results.

```
    Ctrl-V           - like image
    Ctrl-B           - bookmark
    Ctrl-Alt-B       - bookmark privately
    Ctrl-Shift-B     - remove bookmark
    Ctrl-Alt-Shift-M - add bookmark tag
    Ctrl-F           - follow
    Ctrl-Alt-F       - follow privately
    Ctrl-Shift-F     - unfollow
```

**Comment by Pandaa-P** [comment]
Thank you from the bottom of my heart, thank you so much
I have such deep respect for you, I am aware that this took a huge amount of your time
Really really appreciate your amazing work

**Comment by Pandaa-P** [comment]
I really appreciate that you've taken time to inform me of that. I updated the script and enabled the option and everything work like charm. Thank you.

>It's off by default since most people don't need it 

I agree, most people like to open each image in a tab and see the full res, but for people who bookmark soooo many images and references [ image hoarders like me] this option is important, because opening every image in a tab then bookmarking it is a waste of resource, time and just pain in the ass.

 Thank you so much for making my hoarder life easy.

**Comment by ppixiv** [comment]
You can bookmark from searches either way by right clicking thumbnails, the only difference is pressing ctrl doesn't open the menu by default.

**Comment by Pandaa-P** [comment]
Yeah I know, but the right click wasn't working for me, and the menu vanish the moment my finger make a slight move wasn't helping, it was is just too fast. I can't even point the mouse pointer to click the heart sign as I told you before. using another key to trigger the menu and right click to select an option is far more convenient. Theoretically right clicking would be superior but unlike Windows system or Browsers Context Menu which stick until the user click on something this script Context Menu vanish the moment my finger slightly move let alone clicking on the heart sign.
